### PR TITLE
Adds Support to Search through URL pattern of SIT data over a TimeRange

### PR DESCRIPTION
**Pull request by ankitkmr**

### CODE REVIEW

**Comment by ankitkmr** [comment]
@dpshelio My first PR :D Let me know your comments. I've added the get_url... routine to return a list of url for all the dates in the time range. It got a bit complex but the url pattern is a bit strange.

File: sunpy/lightcurve/sources/sit.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+         tuple containing specified start and end dates .
+
+        Parameters
+        ----------
+        args : TimeRange, datetimes, date strings
+            Date range should be specified using a TimeRange, or start
+            and end dates at datetime instances or date strings.
+
+        instrument_orientation : string
+            SIT instrument orientation (default = "ahead")
```
Comment (dpshelio) [inline sunpy/lightcurve/sources/sit.py]: I would not call this `orientation` as it relates to the spacecraft where it is.  [STEREO](http://www2.physik.uni-kiel.de/stereo/downloads/STEREO_orbit.png) is a mission with two spacecraft, one is ahead of earth and the other behind, therefore the name.  I would call this `stereo_spacecraft` or something like that.

File: sunpy/lightcurve/sources/sit.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        base_url = 'http://www.srl.caltech.edu/STEREO/DATA/SIT/' + instrument_orientation + '/' + type_of_average + '/' 
+
+        if type_of_average == '1min' or type_of_average == '10min':
+            base_url = base_url + atomic_specie + '/'
+        
+        #adding the file name to base url
+        if instrument_orientation == "ahead":
+            base_url = base_url + 'SIT_Ahead_' + type_of_average + '_' + atomic_specie + '_'
+        elif instrument_orientation == "behind":
+            base_url = base_url + 'SIT_Behind_' + type_of_average + '_' + atomic_specie + '_'
```
Comment (dpshelio) [inline sunpy/lightcurve/sources/sit.py]: I don't think you need this if here, you could do:
```python
base_url = base_url + 'SIT'+ '_' + instrument_orientation.capitalize() + '_' + type_of_average + '_' + atomic_specie + '_'
``` 
or too:
```python
base_url = '_'.join([base_url', 'SIT',  instrument_orientation.capitalize(), type_of_average, atomic_specie, ''])
```

File: sunpy/lightcurve/sources/sit.py (hunk 1 above)
Comment (ankitkmr) [inline sunpy/lightcurve/sources/sit.py]: Done changed to stereo_spacecraft everwhere.

File: sunpy/lightcurve/sources/sit.py (hunk 2 above)
Comment (ankitkmr) [inline sunpy/lightcurve/sources/sit.py]: changed. Thanks !!

**Pull request closed without merging**